### DISCUSSION TITLE
Failed Tests of Metrics Toolkit in Anypoint Studio

### DISCUSSION THREAD

**Discussion by dmAP99** [mulesoft-catalyst/metrics-toolkit#145] [Q&A] [answered]
Hi.  I deployed Mulesoft Metrics Tooklit locally (Anypoint Studio) and doing a test using the API Console. I'm having issues with the test. Any assistance is much appreciated. 

1) GET http://localhost:8081/api/platform-metrics in API Console is not showing the "Try it' button 

2) Tried POST http://localhost:8081/api/platform-metrics/load with Headers (X-ANYPNT-ORG-ID X-ANYPNT-CLIENT-ID X-ANYPNT-CLIENT-SECRET X-ANYPNT-USERNAME X-ANYPNT-PASSWORD from Anypoint Platform) including Bearer Token but getting errors below.  

3) Does my Anypoint Platform account need to have CloudHub Admin access?

Errors:
    a) Error HTTP POST on resource 'https://anypoint.mulesoft.com:443/accounts/api/v2/oauth2/token' failed: unauthorized (401). Five X-Anypoint headers + Authentication Bearer provided here. 

    b) Error HTTP GET on resource 'https://anypoint.mulesoft.com:443/accounts/api/organizations/xxxxxxxxxxxxxxxxxxxxxxxx/hierarchy' failed: unauthorized (401). Without CLIENT-ID and CLIENT-SECRET here.

**Comment by richardmckinley** [comment, marked as answer]
HI @dmAP99 

1 - I have just used the console to send a request locally without issue. Can you provide a screenshot of the behaviour you're seeing? 
2 - You do not need to provide both platform user credentials (X-ANYPNT-USERNAME X-ANYPNT-PASSWORD) and connected app credentials (X-ANYPNT-CLIENT-ID X-ANYPNT-CLIENT-SECRET) just one set depending on the auth method you are using. 
3 - If you are using a Platform User account to call the platform APIs, then yes, this user needs Org Admin, and CloudHub Admin rites in each business group and environment. Please see the [requirements](https://github.com/mulesoft-catalyst/metrics-toolkit#requirements) section in the main docs.

**Reply by MFredX** [reply]
@richardmckinley would the same approach work if our organisation has implemented external identity management (SAML in my case)?

I have trouble using the the tool and I am thrown a 500 server error when I run the app and enter my credentials on the /platform-metrics console page;

`
{
"message": "Internal Server error",
"error": "HTTP POST on resource 'https://anypoint.mulesoft.com:443/accounts/login' failed: unauthorized (401)."
}
`

**Reply by richardmckinley** [reply]
HI @MFredX, 
No the toolkit does not implement a SAML auth flow, it uses unfederated credentials. If your org has external identity management implemented you would need to follow the token generation process described [here](https://docs.mulesoft.com/access-management/saml-bearer-token) and extend the toolkit accordingly. 
Hope that helps, 
Richard
